lib/pymedphys/_trf/decode/header.py

```diff
@@ -38,34 +52,62 @@ def determine_header_length(trf_contents: bytes) -> int:
     """
 
-    column_end = b"\t\xdc\x00\xe8\t\xdc\x00\xe9\t\xdc\x00\xea\t\xdc\x00\xeb\t\xdc\x00"
-    header_length = trf_contents.index(column_end) + len(column_end)
+    # Date regex match (any 25 characters \x00-\x19) with pattern (\d\d/\d\d/\d\d \d\d:\d\d:\d\d Z)
+    # UTC offset regex (any 25 characters \x00-\x19) with pattern ([\+|\-]\d\d:\d\d)
+    # Treatment name regex match (any 50 characters \x00-\x32) with pattern ([\x20-\x7F]*) according to ASCII table encoding
+    # Machine number (any 10 characters \x00-\x10) with pattern ([\x20-\x7F]*) according to ASCII table encoding
+    # Treatment Record related header
+
+    regex_trf = (
+        rb"[\x00-\x19]"  # start bit
+        + rb"(\d\d[/|-]\d\d[/|-]\d\d \d\d:\d\d:\d\d Z)"  # date
+        + rb"[\x00-\x19]"  # divider bit
+        + rb"([\+|\-]\d\d:\d\d)"  # time zone
+        + rb"[\x00-\x32]"  # divider bit
+        + rb"([\x20-\x7F]*)"  # field label and name
+        + rb"[\x00-\x10]"  # divider bit
+        + rb"([\x20-\x7F]*)"  # machine name
+        + rb"([\x00-\xFFF]*)"  # divider bit
+    )
+
+    # The first 4 groups in the regex match are dynamically allocated as
+    # depend on the date, timezone, field and machine
+    # The field is likely to change the number of bytes.
+
+    start_header_length = 0
+    match = re.match(regex_trf, trf_contents)
+
+    if match:
+        groups = match.groups()
+        start_header_length = match.span(4)[1]
+
+    else:
+        raise ValueError("Unexpected header content found")
 
-    # test = trf_contents.split(b"\t")
-    # row_skips = 6
-    # i = next(i for i, item in enumerate(test[row_skips::]) if len(item) > 3) + row_skips
-    # header_length_old_method = len(b"\t".join(test[0:i])) + 3
+    item_parts_number = np.frombuffer(groups[4][12:16], dtype=np.int32).item()
+    final_header_length = len(groups[4][0 : 16 + 4 * item_parts_number])
 
-    # if header_length_old_method != header_length:
-    #     raise ValueError("Inconsistent header length determination")
+    header_length = start_header_length + final_header_length
 
     return header_length
 
 
 def _header_match(contents):
-    match = re.match(
+
+    regex_trf = (
         rb"[\x00-\x19]"  # start bit
-        rb"(\d\d[/-]\d\d[/-]\d\d \d\d:\d\d:\d\d Z)"  # date
-        rb"[\x00-\x19]"  # divider bit
-        rb"((\+|\-)\d\d:\d\d)"  # time zone
-        rb"[\x00-\x25]"  # divider bit
-        rb"([\x20-\x7F]*)"  # field label and name
-        rb"[\x00-\x19]"  # divider bit
-        rb"([\x20-\x7F]+)"  # machine name
-        rb"[\x00-\x19]",  # divider bit
-        contents,
+        + rb"(\d\d[/|-]\d\d[/|-]\d\d \d\d:\d\d:\d\d Z)"  # date
+        + rb"[\x00-\x19]"  # divider bit
+        + rb"([\+|\-]\d\d:\d\d)"  # time zone
+        + rb"[\x00-\x32]"  # divider bit
+        + rb"([\x20-\x7F]*)"  # field label and name
+        + rb"[\x00-\x10]"  # divider bit
+        + rb"([\x20-\x7F]*)"  # machine name
+        + rb"([\x00-\xFFF]*)"  # divider bit
```

Comment:
Would this be correct here?

```suggestion
        + rb"([\x00-\xFFF]*)"  # mu, version, and item parts numbers
```

Reply:
Does this \xFFF* end up just picking up the rest of the entire file?

Reply:
because we could not identify the bit divider that separates the header from the remaining of content i used this. then with the information extracted we calculate the header length. this should return the remainder of the file.

Reply:
> this should return the remainder of the file.

kk, potentially then the regex for that line should just be `.*`?